### PR TITLE
NEW: Add `flat` function and tests

### PR DESCRIPTION
**Pull request by r-hyndman**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -14,7 +14,36 @@ function filter(array, callback) {
...
+};
+
+function findIndex(array, callback) {
+  for (let i = 0; i < array.length; ++i) {
+    if (callback(array[i], i, array)) return i;
+  }
+  return -1;
+};
+
+function flat(array, depth) {
```
Comment (reubsinit) [inline src/index.js]:
```suggestion
function flat(array, depth = 1) {
```
The depth level specifying how deep a nested array structure should be flattened. Defaults to 1.

File: src/index.js [hunk 2]
```diff
@@ -14,7 +14,36 @@ function filter(array, callback) {
...
+  let result = [];
+  for (let i = 0; i < array.length; ++i) {
+    if (Array.isArray(array[i]) && depth > 0) {
+      result = result.concat((flat(array[i], depth - 1)));
+    } else {
+      result.push(array[i]);
+    }
+  }
```
Comment (reubsinit) [inline src/index.js:39]: This is a really nice solution!

**Review by reubsinit — CHANGES_REQUESTED** [review]
Your `flat` is 🔥 - one small change requested

File: src/index.js (hunk 2 above)
Comment (r-hyndman) [reply src/index.js:39]: Who's best manatee now, huh? 🥇

**Review by reubsinit — APPROVED** [review]

**Pull request merged**